src/core/dbus-unit.c

```diff
@@ -722,6 +723,53 @@ int bus_unit_method_clean(sd_bus_message *message, void *userdata, sd_bus_error
         return sd_bus_reply_method_return(message, NULL);
 }
 
+static int bus_unit_method_freezer_generic(sd_bus_message *message, void *userdata, sd_bus_error *error, FreezerAction action) {
+        const char* perm;
```

Comment:
```
    archlinux_systemd_ci: ../src/core/dbus-unit.c: In function ‘bus_unit_method_freezer_generic’:
    archlinux_systemd_ci: ../src/core/dbus-unit.c:727:21: error: variable ‘perm’ set but not used [-Werror=unused-but-set-variable]
    archlinux_systemd_ci:   727 |         const char* perm;
    archlinux_systemd_ci:       |                     ^~~~
    archlinux_systemd_ci: cc1: all warnings being treated as errors
```

Reply:
This is caused by `mac_selinux_unit_access_check()` being a macro, which is optimized away if we compile without SELinux support (which we do):

https://github.com/systemd/systemd/blob/9d78dd1cce0f7f78153e38c6e867f31e0eb347b3/src/core/selinux-access.h#L11-L24

So, basically, the `r = mac_selinux_unit_access_check(u, message, perm, error);` line expands to `r = 0;`, which effectively makes the `perm` variable to be set, but not used.

Reply:
I've pushed additional commit which removes the preprocessor check in the header as it is not really needed and as a consequence, the warning should go away.